src/experimentation-opportunities/experimentation-opportunities.js

```diff
@@ -96,6 +99,28 @@ export function getRecommendations(lambdaResult) {
   return recommendations;
 }
 
+/* c8 ignore start */
+async function getPresignedUrl(fileName, context, url, site) {
+  const { log } = context;
+  const screenshotPath = `${getS3PathPrefix(url, site)}/${fileName}`;
+  try {
+    const s3Client = AWSXray.captureAWSv3Client(new S3Client({ region: process.env.AWS_REGION }));
+    log.info(`Generating presigned URL for ${screenshotPath}`);
+    const command = new GetObjectCommand({
+      Bucket: process.env.S3_BUCKET_NAME,
+      Key: screenshotPath,
+    });
+    const signedUrl = await getSignedUrl(s3Client, command, {
+      expiresIn: 7 * 24 * 60 * 60, // 7 days in seconds
```

Comment:
*nitpicking*: this could be extracted as a constant and evaluated once at the top of the file